catalog/security/catalog-security-resourceplugin/src/main/java/org/codice/ddf/catalog/security/ResourceUriPolicy.java

```diff
@@ -0,0 +1,200 @@
+                metacardFromCatalog.getResourceURI())) {
+            return new PolicyResponseImpl(null,
+                    Permissions.parsePermissionsFromString(getUpdatePermissions()));
+        }
+
+        return new PolicyResponseImpl();
+    }
+
+    private boolean requiresPermissionsToChangeFrom(URI input, URI catalog) {
+
+        return !uriToString(input).equals(uriToString(catalog));
+    }
+
+    private String uriToString(URI uri) {
+        return uri == null ? "" : uri.toString();
+    }
+
+    @Override
+    public PolicyResponse processPreDelete(List<Metacard> metacards,
+            Map<String, Serializable> properties) throws StopProcessingException {
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostDelete(Metacard input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreQuery(Query query, Map<String, Serializable> properties)
+            throws StopProcessingException {
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostQuery(Result input, Map<String, Serializable> properties)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPreResource(ResourceRequest resourceRequest)
+            throws StopProcessingException {
+
+        return new PolicyResponseImpl();
+    }
+
+    @Override
+    public PolicyResponse processPostResource(ResourceResponse resourceResponse, Metacard metacard)
+            throws StopProcessingException {
+        return new PolicyResponseImpl();
+    }
+
+    public CatalogFramework getCatalogFramework() {
+        return catalogFramework;
+    }
+
+    public void setCatalogFramework(CatalogFramework catalogFramework) {
+        this.catalogFramework = catalogFramework;
+    }
+
+    public String[] getCreatePermissions() {
+        return createPermissions == null ? null : createPermissions.clone();
+    }
+
+    public void setCreatePermissions(String[] createPermissions) {
+        this.createPermissions = createPermissions == null ? null : createPermissions.clone();
+    }
+
+    public FilterBuilder getFilterBuilder() {
+        return filterBuilder;
+    }
+
+    public void setFilterBuilder(FilterBuilder filterBuilder) {
+        this.filterBuilder = filterBuilder;
+    }
+
+    public String[] getUpdatePermissions() {
+        return updatePermissions == null ? null : updatePermissions.clone();
+    }
+
+    public void setUpdatePermissions(String[] updatePermissions) {
+        this.updatePermissions = updatePermissions == null ? null : updatePermissions.clone();
+    }
+
+    protected Metacard getMetacardFromCatalog(String id) throws StopProcessingException {
+        Filter filter = filterBuilder.attribute(Metacard.ID)
```

Comment:
You shouldn't need to query the catalog on an update. The metacard(s) getting updated have all ready been pulled from the catalog and stored in passed in properties under the key Constants.OPERATION_TRANSACTION_KEY. The OperationTransaction object holds the current metacards.